### PR TITLE
Changed default casefolding method to str.casefold; User-defined casefolding

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: docs/intro.rst [hunk 1]
```diff
@@ -34,6 +34,54 @@ for being case-insensitive, of course). This includes the ``clear()`` and
...
+is implemented in its :meth:`~nocaselist.NocaseList.__casefold__` method. That
+method returns the casefolded value for the case-insensitive list items
+(or `None` if the key is `None`).
+
+The default implementation of the :meth:`~nocaselist.NocaseList.__casefold__`
+method calls :meth:`py:str.casefold` on Python 3 and :meth:`py2:str.lower` on
+Python 2. The :meth:`py:str.casefold` method implements the casefolding
+algorithm described in :term:`Default Case Folding in The Unicode Standard`.
+
+In some situations, it may be necessary to change the case-insensitive
```
Comment (KSchopmeyer) [inline docs/intro.rst:58]: Could change to "If it is necessary"

**Review by KSchopmeyer — COMMENTED** [review]
Couple of general comments.

1. In changes.rst and commit message. state reason for change (same reason as for nocasedict)


Other than these comments approve.

**Comment by andy-maier** [comment]
I resolved Karl's comments on this PR and also applied his review comments from PR https://github.com/pywbem/nocasedict/pull/128 to this PR and resolved those as well.

**Comment by andy-maier** [comment]
I'll keep this PR open until PR https://github.com/pywbem/nocasedict/pull/128 gets merged.

**Pull request merged**